sklearn/tree/_classes.py

```diff
@@ -317,6 +317,13 @@ def fit(self, X, y, sample_weight=None, check_input=True,
             raise ValueError("min_impurity_decrease must be greater than "
                              "or equal to 0")
 
+        # TODO: Remove in v0.26
+        if X_idx_sorted != "deprecated":
+            warnings.warn("The parameter 'X_idx_sorted' is deprecated and has "
+                          "no effect. It will be removed in v0.26. You can "
+                          "suppress this warning by not passing any value to "
+                          "the 'X_idx_sorted' parameter.", DeprecationWarning)
```

Comment:
```suggestion
                          "the 'X_idx_sorted' parameter.", FutureWarning)
```

We actually use `FutureWarning` instead of `DeprecationWarning` to be consistent with the pydata ecosystem and do not modify the warning configuration.

Reply:
Thanks for the suggestion!

I will keep in mind for the future.